### DISCUSSION TITLE
Incorrect result on BigNumber subtraction

### DISCUSSION THREAD

**Discussion by brandoncollins7** [josdejong/mathjs#3069] [Q&A]
math.subtract("315051884410500000", "358209676795500000")  = -4315779238500060

The correct answer should be: -43157792385000000. Why would this be happening? This can be replicated using the math.subtract method on the mathjs website via the chrome debugging console.

**Comment by josdejong** [comment]
Thanks for your question @brandoncollins7. I think that your expectation is that `math.subtract` will turn input into BigNumbers when the global config is set to BigNumber. This is a logical reasoning but unfortunately this is not how mathjs works right now. In your example, the strings are turned into regular numbers and then evaluated.

You're not the first to stumble upon this, see: #864. I will mark this as a duplicate, let's continue any discussion over there.

**Comment by harrysarson** [comment]
A good first step might be to throw an error when these functions are called with strings rather than silently converting to numbers.

**Comment by josdejong** [comment]
I'm not sure about that, it is currently a feature that strings are converted into numbers.

**Comment by harrysarson** [comment]
Oh I see, so we want to keep converting strings to some numeric type. But allow what type that is to be configured?

**Comment by josdejong** [comment]
It makes sense to have string input converted to a number type (number, BigNumber, Fraction) depending on the config option `number`, that is exactly what this option doing so far: in cases where mathjs can't know the desired number type for an output, it uses `config.number`. That is uses in the expression parser and in functions like `ones`.

I don't think it will be trivial to implement such a change though.

**Comment by gwhitney** [comment]
I think this is a common question, so moving it to a Q&A discussion.
